File: src/1/1.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

let pos = 50, part1 = 0, part2 = 0;
console.log("Go!");
rl.on('line', (line) => {
        const rot = line.slice(1) * (line[0] == "L" ? -1 : 1)
        const nPos = (pos + rot + 100) % 100;
        part1 += nPos === 0;
        part2 += rot > 0
            ? Math.floor((pos + rot) / 100) - Math.floor(pos / 100)
            : Math.floor((pos - 1) / 100) - Math.floor((pos - 1 + rot) / 100)
        pos = nPos;
});

rl.on("close", () => {
    console.log(part1, part2)
})

File: src/01/1.go
package main

import (
	"fmt"
	"strconv"
	"bufio"
	"os"
)

func main() {
	input := bufio.NewScanner(os.Stdin)
	pos := 50
	part1 := 0
	part2 := 0
	for input.Scan() {
	line := input.Text()
	val, _ := strconv.Atoi(line[1:])
	rot := val
	if line[0] == 'L' {
		rot = -val
	}
	nPos := (pos + rot + 100) % 100
	if nPos == 0 {
		part1++
	}
	if rot > 0 {
		part2 += floorDiv(pos+rot, 100) - floorDiv(pos, 100)
	} else {
		part2 += floorDiv(pos-1, 100) - floorDiv(pos-1+rot, 100)
	}

	pos = nPos
	}
	fmt.Println(part1, part2)
}

func floorDiv(a, b int) int {
	quotient := a / b
	remainder := a % b
	if remainder < 0 {
		quotient--
	}
	return quotient
}